src/main/java/uk/ac/ebi/eva/utils/VcfNumberOfLinesEstimator.java

```diff
@@ -0,0 +1,137 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.utils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import uk.ac.ebi.eva.pipeline.listeners.StepProgressListener;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.zip.GZIPInputStream;
+
+/**
+ * Estimate the number of lines in a VCF file.
+ * <p>
+ * Given that the VCF file could be VERY big then we estimate the number of lines using the following steps:
+ * 1) retrieve the size in bytes of the whole zipped VCF (vcfFileSize)
+ * 2) retrieve the size in bytes of the head of the VCF in a zipped file (vcfHeadFileSize)
+ * 3) retrieve the size in bytes of the first 100 lines of the VCF in a zipped file and divide the value by 100 to
+ * obtain the size of a single line (singleVcfLineSize)
+ * 4) calculate the (vcfFileSize - vcfHeadFileSize) / singleVcfLineSize to estimate the total number of line in the VCF.
+ * <p>
+ * Why 100 in point 3?
+ * Tested on a VCF with 157049 lines, 100 is the best and minimum number of lines to compress. This should generate an
+ * estimated total number of lines similar to the real one.
+ * <p>
+ * In case of small VCF the NUMBER_OF_LINES will be 0 and no % will be printed in {@link StepProgressListener#afterChunk}
+ */
+public class VcfNumberOfLinesEstimator {
+    private static final Logger logger = LoggerFactory.getLogger(VcfNumberOfLinesEstimator.class);
+
+    private static final int NUMBER_OF_LINES = 100;
+
+    public int estimateVcfNumberOfLines(String vcfFilePath) {
+        logger.debug("Estimating the number of lines in the VCF file {}", vcfFilePath);
+        int estimatedTotalNumberOfLines = 0;
+
+        String vcfHead;
+        String vcfSection;
+
+        try {
+            vcfHead = retrieveVcfHead(vcfFilePath);
+            vcfSection = retrieveVcfSection(vcfFilePath);
+        } catch (IOException e) {
+            throw new RuntimeException("Error reading VCF " + vcfFilePath, e);
+        }
+
+        if (!vcfSection.isEmpty()) {
+            File vcfHeadFile;
+            File vcfSectionFile;
+
+            try {
+                vcfHeadFile = FileUtils.newGzipFile(vcfHead, "vcfHeadFile");
+                vcfSectionFile = FileUtils.newGzipFile(vcfSection, "vcfSectionFile");
+            } catch (IOException e) {
+                throw new RuntimeException("Error while creating zip file", e);
+            }
+
+            int vcfFileSize = (int) new File(vcfFilePath).length();
+            int singleVcfLineSize = (int) (vcfSectionFile.length() / NUMBER_OF_LINES);
+            int vcfHeadFileSize = (int) vcfHeadFile.length();
+
+            estimatedTotalNumberOfLines = ((vcfFileSize - vcfHeadFileSize) / singleVcfLineSize);
+        }
+
+        logger.info("Estimated number of lines in VCF file: {}", estimatedTotalNumberOfLines);
```

Comment:
I suggest using `Approximate number of lines in VCF file that contain variants` for 2 reasons:

* Some files have hundreds of header lines and this will look weird if we don't make clear that the header is not included
* People can get a bit paranoid if we don't "estimate" the exact number, because that could mean we are losing some